File: Backend/Travle.Model/Responses/PaymentResponse.cs
namespace Travle.Model.Responses
{
    /// <summary>
    /// A payment as shown on the admin payments screen. Reference fields are flattened to names (traveler,
    /// tour) — never raw ids on screen. <see cref="Status"/> is the enum name; <see cref="RefundedAmount"/>
    /// is the sum of refunds against this payment. Financial records are read-only (never CRUD-edited).
    /// </summary>
    public class PaymentResponse
    {
        public int Id { get; set; }

        public int BookingId { get; set; }
        public string TravelerName { get; set; } = string.Empty;
        public string TravelerUsername { get; set; } = string.Empty;
        public string TourName { get; set; } = string.Empty;

        public decimal Amount { get; set; }
        public string Currency { get; set; } = "bam";

        public decimal PlatformFeePercentage { get; set; }
        public decimal PlatformFeeAmount { get; set; }

        /// <summary>Pending / Succeeded / Failed / Refunded / PartiallyRefunded — the enum name.</summary>
        public string Status { get; set; } = string.Empty;

        /// <summary>Total refunded against this payment (0 when none); and how many refunds.</summary>
        public decimal RefundedAmount { get; set; }
        public int RefundCount { get; set; }

        /// <summary>
        /// True when this captured payment still owes money back and carries no refund yet, because an
        /// automatic attempt failed. Two cases qualify: a cancelled booking, which owes the obligation it
        /// recorded, and a charge captured against a booking that was never honoured at all (expired, or
        /// refused by the amount guard), which owes the whole amount. Drives the admin "Retry refund"
        /// action, and is resolved by the same rule that endpoint enforces.
        /// </summary>
        public bool RefundOwed { get; set; }

        /// <summary>
        /// What a retry will pay, to the fening: the figure the booking recorded when it was cancelled, or
        /// the whole captured charge when there was no cancellation to record one. Present for any cancelled
        /// booking (whether or not the refund went through) so an owed refund reads as a concrete figure
        /// rather than an open question. Null when nothing is or was owed.
        /// </summary>
        public decimal? RefundOwedAmount { get; set; }

        /// <summary>The percentage behind <see cref="RefundOwedAmount"/>; null on the same terms.</summary>
        public int? RefundOwedPercentage { get; set; }

        /// <summary>
        /// Who or what cancelled the booking (the enum name) — the reason that percentage is what it is.
        /// Null for a booking that was never cancelled.
        /// </summary>
        public string? CancellationSource { get; set; }

        /// <summary>When the booking was cancelled; null if it never was.</summary>
        public DateTime? CancelledAt { get; set; }

        public DateTime? SucceededAt { get; set; }
        public DateTime CreatedAt { get; set; }
    }
}

File: UI/travle_core/lib/src/models/payment_response.dart
import 'package:json_annotation/json_annotation.dart';

part 'payment_response.g.dart';

/// A payment as shown on the admin payments screen (mirrors the backend
/// `PaymentResponse`). Reference fields are flattened to names (traveler, tour).
/// [status] is the enum name; [refundedAmount] is the total refunded against it.
@JsonSerializable()
class PaymentResponse {
  PaymentResponse({
    required this.id,
    required this.bookingId,
    required this.travelerName,
    required this.travelerUsername,
    required this.tourName,
    required this.amount,
    required this.currency,
    required this.platformFeePercentage,
    required this.platformFeeAmount,
    required this.status,
    required this.refundedAmount,
    required this.refundCount,
    required this.createdAt,
    this.succeededAt,
  });

  final int id;
  final int bookingId;
  final String travelerName;
  final String travelerUsername;
  final String tourName;

  final double amount;
  final String currency;

  final double platformFeePercentage;
  final double platformFeeAmount;

  /// Pending / Succeeded / Failed / Refunded / PartiallyRefunded.
  final String status;

  final double refundedAmount;
  final int refundCount;

  final DateTime? succeededAt;
  final DateTime createdAt;

  factory PaymentResponse.fromJson(Map<String, dynamic> json) =>
      _$PaymentResponseFromJson(json);

  Map<String, dynamic> toJson() => _$PaymentResponseToJson(this);
}
